lib/provider/local/index.js

```diff
@@ -73,6 +75,32 @@ exports.validate = function(skew, request) {
   }
 };
 
+const digest =
+
+/**
+ * Validate correctness of the request's digest header
+ *
+ * @param  {String} algorithm The configured hashing algorithm
+ * @param  {HTTP.IncomingMessage} request The request to validate
+ *
+ * @throws {Errors.AuthorizationError}    If the digest header does not match the request body
+ */
+exports.digest = function(algorithm, request) {
+  const body = request.body.toString();
+  const digestHeader = request.headers.digest;
+
+  const hash = Crypto.createHash(algorithm);
+
+  hash.update(body);
+  const signature = hash.digest('base64');
+
+  if (digestHeader !== `${algorithm}=${signature}`) {
```

Comment:
i dont think constant time comparison is required here as there is no secret involved